diff --git a/node_modules/@strapi/utils/dist/import-default.js b/node_modules/@strapi/utils/dist/import-default.js
index 9b8cd6f..17b544d 100644
--- a/node_modules/@strapi/utils/dist/import-default.js
+++ b/node_modules/@strapi/utils/dist/import-default.js
@@ -1,8 +1,16 @@
 'use strict';
 
-/* eslint-disable @typescript-eslint/no-var-requires */ function importDefault(modName) {
-    const mod = require(modName);
-    return mod && mod.__esModule ? mod.default : mod;
+/* eslint-disable @typescript-eslint/no-var-requires */ function importDefault(modName, req) {
+    if (typeof __webpack_require__ === 'function') {
+        if (!modName.startsWith('./')){
+            modName = `./${modName}`;
+        }
+        const mod = req(modName);
+        return mod && typeof mod === 'object' && 'default' in mod ? mod.default : mod;
+    } else {
+        const mod = require(modName);
+        return mod && mod.__esModule ? mod.default : mod;
+    }
 }
 
 module.exports = importDefault;
diff --git a/node_modules/@strapi/utils/dist/index.mjs.rej b/node_modules/@strapi/utils/dist/index.mjs.rej
new file mode 100644
index 0000000..ebb6bd6
--- /dev/null
+++ b/node_modules/@strapi/utils/dist/index.mjs.rej
@@ -0,0 +1,22 @@
+diff a/node_modules/@strapi/utils/dist/index.mjs b/node_modules/@strapi/utils/dist/index.mjs	(rejected hunks)
+@@ -749,9 +749,17 @@ const createVisitorUtils = ({ data })=>({
+     });
+ var traverseEntity$1 = curry(traverseEntity);
+ 
+-/* eslint-disable @typescript-eslint/no-var-requires */ function importDefault(modName) {
+-    const mod = require(modName);
+-    return mod && mod.__esModule ? mod.default : mod;
++/* eslint-disable @typescript-eslint/no-var-requires */ function importDefault(modName, req) {
++    if (typeof __webpack_require__ === 'function') {
++        if (!modName.startsWith('./')){
++            modName = `./${modName}`;
++        }
++        const mod = req(modName);
++        return mod && typeof mod === 'object' && 'default' in mod  ? mod.default : mod;
++    } else {
++        const mod = require(modName);
++        return mod && mod.__esModule ? mod.default : mod;
++    }
+ }
+ 
+ var machineId = (()=>{
